### PR TITLE
Enable Static Imports Checkstyle

### LINKED ISSUE #1200: Checkstyle: enable AvoidStaticImportCheck
Our list previously:
```
org.awaitility.Awaitility.*,
org.junit.Assert.*,
org.junit.Assume.*,
org.mockito.Mockito.*,
org.mockito.ArgumentMatchers.*,
org.mockito.AdditionalAnswers.*,
org.hamcrest.CoreMatchers.*,
org.hamcrest.Matchers.*,
org.hamcrest.MatcherAssert.*,
org.mockito.hamcrest.MockitoHamcrest.*,
org.hamcrest.collection.IsCollectionWithSize.*,
org.assertj.core.api.Assertions.*,
org.assertj.core.api.Assumptions.*,
org.mockito.BDDMockito.*,
org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*,
org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
```

Part of #1195.

### PR DESCRIPTION
**Pull request by dzou**
This enables the static imports checkstyle check.

Fixes #1200.

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-logging-sample/src/test/java/com.example/LoggingSampleApplicationTests.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ public void testLogRecordedInStackDriver() {
 
 		String logFilter = String.format(LOG_FILTER_FORMAT, traceHeader);
 
-		await().atMost(60, TimeUnit.SECONDS)
+		Awaitility.await().atMost(60, TimeUnit.SECONDS)
```
Comment (artembilan) [inline spring-cloud-gcp-samples/spring-cloud-gcp-logging-sample/src/test/java/com.example/LoggingSampleApplicationTests.java]: Let's ask Spring Cloud guys to include this one into exclude as well!
In that https://github.com/spring-cloud/spring-cloud-build/issues project.

File: spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/it/GcpConfigIntegrationTests.java [hunk 2]
```diff
@@ -27,7 +27,8 @@
 import org.springframework.context.ConfigurableApplicationContext;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assumptions.assumeThat;
```
Comment (artembilan) [inline spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/it/GcpConfigIntegrationTests.java]: May be it's time to ask Spring Cloud team to include this `Assumptions` for static imports as well?

Comment (dzou) [reply spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/it/GcpConfigIntegrationTests.java]: Ok, I have created https://github.com/spring-cloud/spring-cloud-build/pull/85 requesting to add Awaitility and Assumptions to the spring-cloud-build checkstyle.

For now, I'll put this PR on hold and wait to see if the changes are accepted.

Comment (dzou) [reply spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/config/it/GcpConfigIntegrationTests.java]: spring-cloud/spring-cloud-build#85 was just merged; I am going to close this PR and redo PR to reflect those changes.

**Pull request closed without merging**